src/main/java/az/edu/turing/domain/dao/Dao.java

```diff
@@ -0,0 +1,12 @@
+package az.edu.turing.domain.dao;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface Dao <E, T>{
+    E insert(E e);
+    Collection<E> getAll();
+    Optional<E> getById(T id);
+    E deleteById(T id);
+    E update(E e, T id);
```

Comment:
E update(E e); I think, we don't need an ID.

Reply:
> E update(E e); I think, we don't need an ID.

Yes. Because e already has an id and we can use e.id.

Reply:
'create' is better than 'insert' I think.
Use ctrl+alt+L please and one blank line after interface.